spec/system/viewing_sidebar_spec.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+describe "Viewing sidebar", type: :system, js: true do
+  fab!(:admin) { Fabricate(:admin) }
+  fab!(:user) { Fabricate(:user) }
+  fab!(:category_sidebar_section_link) { Fabricate(:category_sidebar_section_link, user: user) }
+
+  describe "when using the legacy navigation menu" do
+    before { SiteSetting.navigation_menu = "legacy" }
+
+    it "should display the sidebar when `enable_sidebar` query param is '1'" do
+      sign_in(user)
+
+      visit("/latest?enable_sidebar=1")
+
+      sidebar = PageObjects::Components::Sidebar.new
+
+      expect(sidebar).to have_category_section_link(category_sidebar_section_link.linkable)
```

Comment:
We could probably use something more general, like checking if `#d-sidebar` or `.sidebar-sections` is visible

Reply:
E.g. `expect(sidebar).to be_visible` can be used if there is a `visible?` method defined on the page object

Reply:
Something like

```
def visible?
  page.has_css?("#d-sidebar")
end
```

Reply:
O I also wanted to check that the sidebar is displaying the right stuff not just that it is present. I've added `expect(sidebar).to be_visible` above.